src/models/instance.ts

```diff
@@ -155,4 +155,39 @@ export default class Instance {
                 : null,
         });
     }
+
+    public async check(): Promise<Response> {
+        const { url, username, password } = this.data;
+        const headers = new Headers({
+            Authorization: "Basic " + btoa(username + ":" + password),
+        });
+
+        try {
+            const res = await fetch(url + "/api/system/info", { method: "GET", headers });
```

Comment:
If we keep it in the models it should be window agnostic to be reused on Phase 2.

Something like this would also work without fetch nor btoa which are not available on node. Note that setting the headers is not needed because axios does it alone but beware that the response object will change a little. 

```js
axios.get(url + "/api/system/info", { auth: {
    username, 
    password, 
  }});
```

Detecting the CORS issue is not possible in axios either, I tried and not succeed so the if conditions you have are good enough.

Reply:
There are some isomorphic (node/browser) fetch packages, but since we are already using axios in this app, I'd go with axios.